src/main/fc/fc_rc.c

```diff
@@ -239,6 +237,105 @@ FAST_CODE FAST_CODE_NOINLINE void processRcCommand(void)
+    uint8_t updatedChannel = 0;
+
+    static float lastRxData[4];
+    static pt1Filter_t rcCommandFilter[4];
+    static bool initialized = false;
+    static bool filterInitialized = false;
+    static float rxFrameTimeSum;
+    static int rxFrameCount;
+    static uint16_t defaultCutoffFrequency;
+    static uint16_t filterCutoffFrequency;
+    static uint16_t derivativeCutoffFrequency;
+    static uint8_t interpolationChannels;
+
+    if (!initialized) {
+        initialized = true;
+        interpolationChannels = rxConfig()->rcInterpolationChannels + 2; //"RP", "RPY", "RPYT"
+        filterCutoffFrequency = rxConfig()->rc_smoothing_input_cutoff;
+        derivativeCutoffFrequency = rxConfig()->rc_smoothing_derivative_cutoff;
+    }
+
+    if (isRXDataNew) {
+        for (int i = 0; i < interpolationChannels; i++) {
+            lastRxData[i] = rcCommand[i];
+        }
+        // If the filter cutoffs are set to auto and we have good rx data, then determine the average rx frame rate
+        // and use that to calculate the filter cutoff frequencies
+        if ((filterCutoffFrequency == 0) || (derivativeCutoffFrequency == 0)) {
+            if (rxIsReceivingSignal()) {
+                rxFrameTimeSum += currentRxRefreshRate;
+                rxFrameCount++;
+                if (rxFrameCount >= RC_SMOOTHING_FILTER_TRAINING_SAMPLES) {
+                    const float avgRxFrameRate = rxFrameTimeSum / rxFrameCount / 1000.0f;
+                    defaultCutoffFrequency = lrintf(RC_SMOOTHING_FILTER_AUTO_HZ / (avgRxFrameRate / RC_SMOOTHING_FILTER_AUTO_MS));
+                    filterCutoffFrequency = (filterCutoffFrequency == 0) ? defaultCutoffFrequency : filterCutoffFrequency;
+                    derivativeCutoffFrequency = (derivativeCutoffFrequency == 0) ? defaultCutoffFrequency : derivativeCutoffFrequency;                
+                }
+            } else {
+                rxFrameTimeSum = 0;
+                rxFrameCount = 0;
+            }
+        }
+    }
+
+    DEBUG_SET(DEBUG_RC_SMOOTHING, 0, lrintf(lastRxData[rxConfig()->rc_smoothing_debug_axis]));
+    DEBUG_SET(DEBUG_RC_SMOOTHING, 3, defaultCutoffFrequency);
+
+    // Once we've determined the filter cutoff frequencies then initialize the filters
+    if (!filterInitialized && (targetPidLooptime > 0) && (filterCutoffFrequency != 0) && (derivativeCutoffFrequency != 0)) {
+        const float dT = targetPidLooptime * 0.000001f;
+        for (int i = 0; i < interpolationChannels; i++) {
+            pt1FilterInit(&rcCommandFilter[i], pt1FilterGain(filterCutoffFrequency, dT));
+        }
+        pidInitSetpointDerivativeLpf(derivativeCutoffFrequency, rxConfig()->rc_smoothing_debug_axis);
+        filterInitialized = true;
+    }
+
+    for (updatedChannel = ROLL; updatedChannel < interpolationChannels; updatedChannel++) {
+        if (filterInitialized) {
+                rcCommand[updatedChannel] = pt1FilterApply(&rcCommandFilter[updatedChannel], lastRxData[updatedChannel]);
```

Comment:
I like this more as a ternary operator.
You seem to have indentation off here.

Reply:
It may be even better to init filter to dummy values (alpha = 0) in `initialized == false` and avoid the test here ...

Reply:
`filterCutoffFrequency` and `derivativeCutoffFrequency` are not really dummy values.  They get assigned in the first initialization stage from the configuration parameters.
```
    if (!initialized) {
        initialized = true;
        interpolationChannels = rxConfig()->rcInterpolationChannels + 2; //"RP", "RPY", "RPYT"
        filterCutoffFrequency = rxConfig()->rc_smoothing_input_cutoff;
        derivativeCutoffFrequency = rxConfig()->rc_smoothing_derivative_cutoff;
    }
```
The "training" phase needs to run inside of `if (isRXDataNew) {` because we only want to sum when we receive new rx frames.  Once we reach the required number of samples we compute average and only override the configuration cutoff if it's 0 (0 = AUTO).

I'll rearrange the logic to wrap the entire sum, average and filter initialization inside of `if (!filterInitialized) {`